Source/SpectrumAnalysis/KTSeqTrackFinder.cc

```diff
@@ -0,0 +1,539 @@
+        myNewTrack.SetIntercept( myNewTrack.GetStartFrequency() - myNewTrack.GetSlope() * myNewTrack.GetStartTimeInRunC() );
+
+    }
+
+    void KTSeqTrackFinder::SearchTrueLinePoint(Point& Point, std::vector<double>& slice)
+    {
+        double Delta = fConvergeDelta + 1.0;
+        unsigned loop_counter = 0;
+        double dF = fBinWidth;
+        int max_iterations = 10;
+
+        double frequency = Point.fPointFreq;
+        double amplitude = Point.fAmplitude;
+        unsigned frequencybin = Point.fBinInSlice;
+        double old_frequencybin;
+        
+
+
+        while(std::abs(Delta) > fConvergeDelta and loop_counter < max_iterations)
+        {
+            loop_counter++;
+
+            old_frequencybin = frequencybin;
+
+            if (frequencybin > fMinBin + fSearchRadius and frequencybin < fMaxBin - fSearchRadius)
+            {
+                this-> WeightedAverage(slice, frequencybin, frequency);
+                Delta = std::abs(frequencybin - old_frequencybin);
+            }
+            else
+            {
+                KTDEBUG(stflog, "frequencybin was not in correct range ");
+                Delta = 0.0;
+                amplitude = 0.0;
+            }
+        }
+
+        // Calculate "true" line amplitude in slice by summing neighboring point amplitudes
+        // and set slice amplitude to zero
+        if (frequencybin > fMinBin + fLinePowerWidth and frequencybin < fMaxBin - fLinePowerWidth)
+        {
+            amplitude = 0;
+            for (int iBin = frequencybin - fLinePowerWidth; iBin <= frequencybin + fLinePowerWidth; iBin++)
+            {
+                amplitude += slice[iBin];
+                slice.at(iBin)=fPointAmplitudeAfterVisit;
+            }
+        }
+        else
+        {
+            amplitude = slice[frequencybin];
+            slice.at(frequencybin)=fPointAmplitudeAfterVisit;
+        }
+        // Set larger area to zero if possible
+        if (frequencybin > fMinBin + fMinFreqBinDistance and frequencybin < fMaxBin - fMinFreqBinDistance)
+        {
+            for (int iBin = frequencybin - fMinFreqBinDistance; iBin <= frequencybin + fMinFreqBinDistance; iBin++)
+            {
+                slice.at(iBin)=fPointAmplitudeAfterVisit;
+            }
+        }
+        // if point was to close to edge, try again with smaller range
+        else if (frequencybin > fMinBin + fSearchRadius and frequencybin < fMaxBin - fSearchRadius)
+        {
+            for (int iBin = frequencybin - fSearchRadius; iBin <= frequencybin + fSearchRadius; iBin++)
+            {
+                slice.at(iBin)=fPointAmplitudeAfterVisit;
+            }
+        }
+
+        // Correct values currently still stored in Point
+        Point.fBinInSlice = frequencybin;
+        Point.fPointFreq = frequency;
+        Point.fAmplitude = amplitude;
+    }
+
+
+
+    inline void KTSeqTrackFinder::WeightedAverage(const std::vector<double>& slice, unsigned& FrequencyBin, double& Frequency)
```

Comment:
1. Same comment about variable names for: `FrequencyBin`, `Frequency`, `new_frequencyBin`, `new_Frequency`

1. Same comment about prefixing `++` for `iBin`